daemon/sailjailclient.c

```diff
@@ -0,0 +1,1284 @@
+{
+    if( !self->cli_session_bus ) {
+        GError *err = NULL;
+        self->cli_session_bus = g_bus_get_sync(G_BUS_TYPE_SESSION, NULL, &err);
+        if( !self->cli_session_bus ) {
+            log_err("failed to connect to D-Bus SessionBus: %s", err->message);
+            exit(EXIT_FAILURE);
+        }
+        g_clear_error(&err);
+    }
+    return self->cli_session_bus;
+}
+
+const char *
+client_desktop_exec(client_t *self)
+{
+    return client_get_appinfo_string(self, DESKTOP_KEY_EXEC);
+}
+
+const char *
+client_sailjail_organization_name(client_t *self)
+{
+    return client_get_appinfo_string(self, SAILJAIL_KEY_ORGANIZATION_NAME);
+}
+
+const char *
+client_sailjail_application_name(client_t *self)
+{
+    return client_get_appinfo_string(self, SAILJAIL_KEY_APPLICATION_NAME);
+}
+
+const char **
+client_sailjail_application_permissions(client_t *self)
+{
+    return client_get_appinfo_strv(self, SAILJAIL_KEY_PERMISSIONS);
+}
+
+const char *
+client_maemo_service(client_t *self)
+{
+    return client_get_appinfo_string(self, MAEMO_KEY_SERVICE);
+}
+
+const char *
+client_maemo_method(client_t *self)
+{
+    return client_get_appinfo_string(self, MAEMO_KEY_METHOD);
+}
+
+/* ------------------------------------------------------------------------- *
+ * CLIENT_PROPERTIES
+ * ------------------------------------------------------------------------- */
+
+static bool
+client_is_privileged(const client_t *self)
+{
+    return self->cli_is_privileged;
+}
+
+static void
+client_set_privileged(client_t *self, bool is_privileged)
+{
+    self->cli_is_privileged = is_privileged;
+}
+
+static const gchar **
+client_get_argv(const client_t *self, int *pargc)
+{
+    if( pargc )
+        *pargc = self->cli_argc;
+    return (const gchar **)self->cli_argv;
+}
+
+static void
+client_set_argv(client_t *self, int argc, char **argv)
+{
+    self->cli_argc = argc;
+    self->cli_argv = argv;
+}
+
+static const gchar **
+client_get_granted(const client_t *self)
+{
+    return (const gchar **)self->cli_granted;
+}
+
+static void
+client_set_granted(client_t *self, gchar **granted)
+{
+    g_strfreev(self->cli_granted),
+        self->cli_granted = granted;
+}
+
+static const gchar *
+client_get_desktop_path(const client_t *self)
+{
+    return self->cli_desktop_path;
+}
+
+static void
+client_set_desktop_path(client_t *self, const char *path)
+{
+    change_string(&self->cli_desktop_path, path);
+}
+
+static void
+client_set_appinfo_variant(client_t *self, const char *key, GVariant *val)
+{
+    g_hash_table_insert(self->cli_appinfo, g_strdup(key), g_variant_ref(val));
+}
+
+static GVariant *
+client_get_appinfo_variant(const client_t *self, const char *key)
+{
+    return g_hash_table_lookup(self->cli_appinfo, key);
+}
+
+const char *
+client_get_appinfo_string(const client_t *self, const char *key)
+{
+    const char *value = NULL;
+    GVariant *variant = client_get_appinfo_variant(self, key);
+    if( variant) {
```

Comment:
I'd really say that the space is in wrong place here. Why this code format?

Reply:
Because of this: https://github.com/sailfishos/sailjail/pull/41#discussion_r621912362#47 

BTW, rustlang is better (or just different, YMMV) with this in some ways. :wink:

Reply:
https://www.kernel.org/doc/html/v4.10/process/coding-style.html I like this better.

Reply:
This style actually hurts my brain so much that all I see is error.

Reply:
Mechanically fixed all of these.